clickhouse/base/socket.h

```diff
@@ -60,6 +60,18 @@ class windowsErrorCategory : public std::error_category {
 
 #endif
 
+#if defined(_unix_)
+
+class getaddrinfoErrorCategory : public std::error_category {
```

Comment:
Do you have to make it part of the API, i.e. visible to the users? IMO it would be better to move declaration to .cpp

Reply:
There's no requirement for it to be public. But users will have access to it through `e.code().category()` and nothing to compare it to.